types.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import "fmt"
+
+type Image struct {
```

Comment:
Can we reuse this struct directly from the Kyverno repo?

Reply:
I tried doing that but I got this error in `types_test.go`

```
map[string]"k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration does not implement "k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration (missing method HasSynced)
```

Reply:
Not sure what that is about....were you able to resolve this?

Reply:
Yet..But this was caused by a dependency conflict between some packages in `k8s.io` and `sigs.k8s.io`

I am not using apiutils and only using image utils as `pointer` is not exposed in the current publicly available version of kyverno library. 

Here is the commit: https://github.com/nirmata/kyverno-notation-aws/pull/6/commits/3e7adf08c038727dce96e6a0bae5bfec2bdcbe30

Reply:
@JimBugwadia Can you help me with this, I have tested input of image variables but I haven't tested the entire flow yet. (Since I haven't touched the logic related to verification, I believe there shouldn't be any issues though). 

I am getting this error while testing:
```
call-aws-signer-extension: |
    failed to check deny conditions: failed to substitute variables in condition key: failed to resolve result.verified at path : failed to execute APICall: HTTP 500 Internal Server Error: failed to verify image 844333597536.dkr.ecr.us-west-2.amazonaws.com/kyverno-demo:v1: failed to resolve digest: failed to retrieve credentials: operation error ECR: GetAuthorizationToken, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, no EC2 IMDS role found, operation error ec2imds: GetMetadata, request canceled, context deadline exceeded
```
I believe it is because of IRSA.
I skipped the IRSA steps, as I thought it has been setup for the repository already, is there anything I have to do in my local environment?

Reply:
Yes, this happens if you do not have AWS credentials configured as these are required for accessing AWS Signer and ECR. Please switch to using an AWS cluster with ISRA setup.